### DISCUSSION TITLE
Play a video from YouTube url

### DISCUSSION THREAD

**Discussion by maxlunla** [microsoft/MixedRealityToolkit-Unity#11789] [Q&A]
I just recently started working on the project, which uses the MRTK with HoloLens 2.

First, I used the YouTube API, which is on the Unity Store ([This one](https://assetstore.unity.com/packages/tools/video/youtube-video-player-youtube-api-29704)) Everything is working fine on the editor; I can load and play the video on the video player from the YouTube URL. But when I deploy and run the HoloLens 2, it crashes, forcing me to close the application.

**My question:** Is there another way to play the YouTube video with a URL? Is there any other API I can use? In case I need to call the browser, is there a way to shrink the web browser's window to show only the video's area? or call YouTube's application instead of the web browser?

**Comment by jackmca** [comment]
Have you tried using the built-in Unity Video Player component? You can change the Source from 'Video Clip' to 'URL', this has worked in the past for me for other URLs, but I have never tried it with YouTube videos specifically.

**Reply by maxlunla** [reply]
I have tried using the built-in Unity Video Player component before. On the URL Source form, it can play the URL, which is an absolute path to the video file (.mp4). When I tried to put the YouTube URL, it got an error and couldn't play the YouTube URL.
